test/Multigoal.lean

```diff
@@ -0,0 +1,105 @@
+import Mathlib.Tactic.Basic
+import Mathlib.Tactic.Conv
+import Mathlib.Tactic.Linter.Multigoal
+
+-- The warning generated by `linter.multiGoal` is not suppressed by `#guard_msgs`,
+-- because the linter is run on `#guard_msgs` itself. This is a known issue, see e.g.
+-- https://leanprover.zulipchat.com/#narrow/stream/348111-batteries/topic/unreachableTactic.20linter.20not.20suppressed.20by.20.60.23guard_msgs.60
+-- We jump through an extra hoop here to silence the warning.
+set_option linter.multiGoal false
+
+-- A deactivated linter does nothing.
+example : True := by
+  by_cases 0 = 0
+  exact .intro
+  exact .intro
+
+#guard_msgs(drop warning) in
+set_option linter.multiGoal true in
+/--
+warning: 'exact .intro' leaves 1 goal 'Lean.Parser.Tactic.exact'
+note: this linter can be disabled with `set_option linter.multiGoal false`
+-/
+#guard_msgs in
+example : True := by
+  by_cases 0 = 0
```

Comment:
Trying to articulate when the linter errors gave me an idea about how to phrase the error message.
- I could be nice to say "`by_cases 0 = 0` produced two goals", but that might be hard in general
- Something that *is* feasible is to change the wording of the error, something like
```
warning: there are 2 unclosed goals before 'exact .intro', and one stays unsolved
please focus on the current goal, for instance using `.` (typed as "\.")

Reply:
I also updated the error message, thanks!